xarray/tests/test_variable.py

```diff
@@ -2204,6 +2204,7 @@ def test_copy_index(self):
         super().test_copy_index()
 
     @pytest.mark.xfail
+    @pytest.mark.filterwarnings("ignore:elementwise comparison failed.*:FutureWarning")
```

Comment:
Most warnings that I have encountered were correct and good to have. Like taking the mean of an all-nan array.
Very often also warnings about casting a NaN to int, which again I find useful because some users might not have that in mind.

In this specific case the future warning is also important to have as it will be a breaking change for most codes. Since it is used together with xfail, this test will fail anyway once the new behavior is implemented.